test/test.pl

```diff
@@ -1589,7 +1589,7 @@ sub bam_compare
 
     my $b1 = open_bgunzip($opts, $bam1);
     my $b2 = open_bgunzip($opts, $bam2);
-    do {
+    do {{
```

Comment:
?

Reply:
For using `last` in a `do-while` statement.

Reply:
This doesn't actually work as intended because the `last` is simply ignored.  Of course it eventually gets to the end of the file, but it takes longer than it really needs to.  This can be shown by feeding the function different never-ending streams, like `<(yes)` and `/dev/zero`, which put it into an endless loop.

The best solution is to add `&& !$fail` to the while clause.

Reply:
Yes, that is a better solution.